manim/mobject/three_dimensions.py

```diff
@@ -189,3 +199,427 @@ def generate_points(self):
+        theta = np.arccos(z / r)
+
+        if x == 0:
+            if y == 0:  # along the z axis
+                phi = 0
+            else:  # along the x axis
+                phi = np.arctan(np.inf)
+                if y < 0:
+                    phi += PI
+        else:
+            phi = np.arctan(y / x)
+        if x < 0:
+            phi += PI
+
+        self.rotate(theta, Y_AXIS, about_point=ORIGIN)
+        self.rotate(phi, Z_AXIS, about_point=ORIGIN)
+
+    def set_direction(self, direction):
+        self.direction = direction
+        self._rotate_to_direction()
+
+
+class Line3D(Cylinder):
+    """A cylindrical line, for use in ThreeDScene.
+
+    Examples
+    ---------
+    .. manim:: ExampleLine3D
+        :save_last_frame:
+
+        class ExampleLine3D(ThreeDScene):
+            def construct(self):
+                axes = ThreeDAxes()
+                line = Line3D(start=np.array([0, 0, 0]), end=np.array([2, 2, 2]))
+                self.set_camera_orientation(phi=75 * DEGREES, theta=30 * DEGREES)
+                self.add(axes, line)
+
+    Parameters
+    ---------
+    start : :class:`numpy.array`
+        The start position of the line.
+    end : :class:`numpy.array`
+        The end position of the line.
+    width : :class:`float`
+        The thickness of the line.
+    color : :class:`str`
+        The color of the line.
+    """
+
+    def __init__(self, start=LEFT, end=RIGHT, width=0.02, color=None, **kwargs):
+        self.set_start_and_end_attrs(start, end)
+        super().__init__(
+            height=get_norm(self.vect), radius=width, direction=self.direction, **kwargs
+        )
+        self.shift((self.start + self.end) / 2)
+        if color is not None:
+            self.set_color(color)
+
+    def set_start_and_end_attrs(self, start, end):
+        """Sets the start and end points of the line.
+
+        If either ``start`` or ``end`` are :class:`~.Mobject`s, this gives their centers.
+        """
+        rough_start = self.pointify(start)
+        rough_end = self.pointify(end)
+        self.vect = rough_end - rough_start
+        self.length = get_norm(self.vect)
+        self.direction = normalize(self.vect)
+        # Now that we know the direction between them,
+        # we can the appropriate boundary point from
+        # start and end, if they're mobjects
+        self.start = self.pointify(start, self.direction)
+        self.end = self.pointify(end, -self.direction)
+
+    def pointify(self, mob_or_point, direction=None):
+        if isinstance(mob_or_point, Mobject):
+            mob = mob_or_point
+            if direction is None:
+                return mob.get_center()
+            else:
+                return mob.get_boundary_point(direction)
+        return np.array(mob_or_point)
+
+    def get_start(self):
+        return self.start
+
+    def get_end(self):
+        return self.end
+
+    def get_direction(self):
+        return self.direction
```

Comment:
https://github.com/ManimCommunity/manim/pull/1026/files#r583102854
Please add to Cylinder instead (since this inherits from Cylinder) and is a simple getter
```suggestion
```